MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -45,14 +53,14 @@ public string GetH264Encoder(EncodingJobInfo state, EncodingOptions encodingOpti
 
                 var codecMap = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
                 {
-                    {"qsv",          "h264_qsv"},
-                    {"h264_qsv",     "h264_qsv"},
-                    {"nvenc",        "h264_nvenc"},
-                    {"amf",          "h264_amf"},
-                    {"omx",          "h264_omx"},
-                    {"h264_v4l2m2m", "h264_v4l2m2m"},
-                    {"mediacodec",   "h264_mediacodec"},
-                    {"vaapi",        "h264_vaapi"}
+                    {"qsv",                  hwEncoder + "_qsv"},
+                    {hwEncoder + "_qsv",     hwEncoder + "_qsv"},
```

Comment:
Why are there these _qsv maps doubled? Can we remove those?

Reply:
> Why are there these _qsv maps doubled? Can we remove those?

I'm assuming the original author wanted both those values to map to the correct hw encoder. I think a refactor and using enums would answer/solve these questions (see my response in your other discussion).

> And that _v4l2m2m one also maps to itself. Should probably be a default value or something.

I'm not sure I understand as I don't see a problem here. Presumably the mapping to itself is the correct action so that we don't have to do any special casing in the code below that uses the map.